### ISSUE TITLE
Visualization does not load canvas in Chromium

### ISSUE THREAD

**Issue by moorepants** [pydy/pydy#228]
On master I try running the mass_spring_damper_example and the canvas fails to load. Here is the traceback:

```
Default extension for cell metadata editing loaded.
Raw Cell Format toolbar preset loaded.
Slideshow extension for metadata editing loaded.
Session: kernel_created (2a0f0e5e-e5e8-455d-958e-4055b723cf3d)
Starting WebSockets: ws://localhost:8889/api/kernels/21406cff-4680-4277-8dc8-b487096421dc
Kernel: kernel_connected (21406cff-4680-4277-8dc8-b487096421dc)
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Uncaught SyntaxError: Unexpected token <
Failed to load resource: the server responded with a status of 404 (Not Found)
Failed to load resource: the server responded with a status of 404 (Not Found)
Failed to load resource: the server responded with a status of 404 (Not Found)
Failed to load resource: the server responded with a status of 404 (Not Found)
Failed to load resource: the server responded with a status of 404 (Not Found)
Kernel: kernel_ready (21406cff-4680-4277-8dc8-b487096421dc)
Failed to load resource: the server responded with a status of 404 (Not Found)
'webkitIndexedDB' is deprecated. Please use 'indexedDB' instead.
Uncaught ReferenceError: Sizzle is not defined
[PyDy INFO]: initializing Visualizer
[PyDy INFO]: Found scene desc from URL
[PyDy INFO]: Loading scene JSON file:static/2015-06-13_09:51:13_scene_desc.json
[PyDy INFO]: Activated UI controls
THREE.WebGLRenderer 58
Error creating WebGL context.
Uncaught TypeError: Cannot read property 'getExtension' of null
[PyDy INFO]: Ajax request completed, adding Objects to scene
```

**Comment by oliverlee** [comment]
This is probably a bug with WebGL support in chromium. What does chrome://gpu/ say about WebGL?

**Comment by moorepants** [comment]
It says WebGL is unavailable. Why would it work for Chrome?

**Comment by oliverlee** [comment]
It did when I checked on my machine.

**Comment by moorepants** [comment]
I'll install Chrome and see if that works. Maybe this is something with my computer and not a larger problem.

**Comment by moorepants** [comment]
Just found this: http://askubuntu.com/questions/299345/how-to-enable-webgl-in-chrome-on-ubuntu

**Comment by sahilshekhawat** [comment]
I tried that link but it still doesn't seems to work on chromium. It says that your GPU is not supported even when I pass allow blacklisted flag.

**Comment by oliverlee** [comment]
It works in Chrome for me but not Chromium. Regardless, I don't think this
is something we can do anything about.

**Comment by moorepants** [comment]
Yeh, I couldn't get it working either. I was under the impression that Chrome only differs from Chromium by some branding and the addition of some proprietary plugins for flash and pdf viewing. I guess not! I know this worked with some Chromium version a half a year ago...

I agree that this can't unlikely be fixed for the release.

**Comment by moorepants** [comment]
I'm moving this issue from the 0.3.0 milestone. If we get a solution before SciPy we can post it here. Until then we will have to deal with this.

**Comment by oliverlee** [comment]
This now works for me with a newer version of Chromium:
Version 48.0.2539.0 custom (64-bit)

Appears to be fixed for all versions greater than 43.0.2357.130:
https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/1463598

You may need to:
```
- Open chrome://settings/
- Go to "Show advanced settings..." -> "System" -> "Use hardware acceleration when available"
```
and
```
- Open chrome://flags/
- "Override software rendering list" -> "Enable"
```

You should then see that WebGL is enabled:
```
- Open chrome://gpu/ 
- Graphics Feature Status -> WebGL: Hardware accelerated
```

**Comment by oliverlee** [comment]
This also in the most recent version of Chromium in OSX:
Version 48.0.2541.0 (64-bit)

If you are having problems, try downloading a newer version of Chromium from here: https://www.chromium.org/getting-involved/download-chromium
